src/main/scala/io/iohk/ethereum/App.scala

```diff
@@ -1,58 +1,35 @@
 package io.iohk.ethereum
 
-import scala.concurrent.ExecutionContext.Implicits.global
-import akka.actor.ActorSystem
-import akka.agent._
 import io.iohk.ethereum.blockchain.sync.FastSyncController
-import io.iohk.ethereum.db.components.{SharedLevelDBDataSources, _}
-import io.iohk.ethereum.domain.{Blockchain, BlockchainImpl}
-import io.iohk.ethereum.network.{PeerManagerActor, ServerActor}
-import io.iohk.ethereum.rpc.JsonRpcServer
-import io.iohk.ethereum.utils.{BlockchainStatus, Config, NodeStatus, ServerStatus}
-import io.iohk.ethereum.network._
+import io.iohk.ethereum.network.ServerActor
+import io.iohk.ethereum.nodebuilder.Node
+import io.iohk.ethereum.utils.Logger
 
-object App {
-
-  import Config.{Network => NetworkConfig}
-
-  val nodeKey = loadAsymmetricCipherKeyPair(Config.keysFile)
+import scala.concurrent.Await
+import scala.util.{Failure, Success, Try}
+import scala.language.postfixOps
 
-  val storagesInstance =  new SharedLevelDBDataSources with Storages.DefaultStorages
-  val blockchain: Blockchain = BlockchainImpl(storagesInstance.storages)
+object App {
 
   def main(args: Array[String]): Unit = {
-    val actorSystem = ActorSystem("etc-client_system")
-
-    val nodeStatus =
-      NodeStatus(
-        key = nodeKey,
-        serverStatus = ServerStatus.NotListening,
-        blockchainStatus = BlockchainStatus(Config.Blockchain.genesisDifficulty, Config.Blockchain.genesisHash, 0))
 
-    val nodeStatusHolder = Agent(nodeStatus)
+    new Node with Logger {
 
-    val peerManager = actorSystem.actorOf(PeerManagerActor.props(nodeStatusHolder, Config.Network.peer, blockchain), "peer-manager")
-    val server = actorSystem.actorOf(ServerActor.props(nodeStatusHolder, peerManager), "server")
-
-    server ! ServerActor.StartServer(NetworkConfig.Server.listenAddress)
+      def tryAndLogFailure(f: () => Any): Unit = Try(f()) match {
+        case Failure(e) => log.warn("Error while shutting down...", e)
+        case Success(_) =>
+      }
 
-    if(Config.Network.Rpc.enabled) JsonRpcServer.run(actorSystem, blockchain, Config.Network.Rpc)
+      override def shutdown(): Unit = {
+        tryAndLogFailure(() => storagesInstance.dataSources.closeAll)
+        tryAndLogFailure(() => Await.ready(actorSystem.terminate, shutdownTimeoutDuration))
+      }
 
-    val fastSyncController = actorSystem.actorOf(
-      FastSyncController.props(
-        peerManager,
-        nodeStatusHolder,
-        blockchain,
-        storagesInstance.storages.mptNodeStorage),
-      "fast-sync-controller")
+      server ! ServerActor.StartServer(networkConfig.Server.listenAddress)
```

Comment:
Yes, but you're right, it's OK. I keep getting confused about Scala initialisation order which is why I don't like having it split over multiple traits.